### PR TITLE
crimson/os/seastore: use different segment seq allocator from journal and ool segments

### PR DESCRIPTION
**Pull request by xxhdx1985126**
Fixes: https://tracker.ceph.com/issues/55143
Signed-off-by: Xuehan Xu <xxhdx1985126@gmail.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/crimson/os/seastore/cache.cc [hunk 1]
```diff
@@ -1389,10 +1389,12 @@ Cache::get_next_dirty_extents_ret Cache::get_next_dirty_extents(
        i != dirty.end() && bytes_so_far < max_bytes;
        ++i) {
     auto dirty_from = i->get_dirty_from();
+#ifndef NDEBUG
     if (!(dirty_from != JOURNAL_SEQ_NULL &&
                 dirty_from != JOURNAL_SEQ_MAX &&
                 dirty_from != NO_DELTAS))
-      ERRORT("{}", *i);
+      ERRORT("{}", t, *i);
```
[the remark this answers is not in the document]
Comment (xxhdx1985126) [reply src/crimson/os/seastore/cache.cc:1396]: yes

File: src/crimson/os/seastore/segment_seq_allocator.h [hunk 2]
```diff
@@ -6,6 +6,13 @@
 #include "crimson/os/seastore/logging.h"
 #include "crimson/os/seastore/seastore_types.h"
 
+#define INIT_JOURNAL_SEGMENT_SEQ 0x00000000
+#define INIT_OOL_SEGMENT_SEQ 0x80000000
```
[the remark this answers is not in the document]
Comment (xxhdx1985126) [reply src/crimson/os/seastore/segment_seq_allocator.h:10]: ah, yes

**Comment by xxhdx1985126** [comment]
@cyx1231st Done:-)

File: src/crimson/os/seastore/cache.cc [hunk 3]
```diff
@@ -1389,13 +1389,14 @@ Cache::get_next_dirty_extents_ret Cache::get_next_dirty_extents(
...
-      ERRORT("{}", *i);
-    ceph_assert(dirty_from != JOURNAL_SEQ_NULL &&
-                dirty_from != JOURNAL_SEQ_MAX &&
-                dirty_from != NO_DELTAS);
+                dirty_from != NO_DELTAS))) {
+      ERRORT("{}", t, *i);
+      ceph_assert(dirty_from != JOURNAL_SEQ_NULL &&
+		  dirty_from != JOURNAL_SEQ_MAX &&
+		  dirty_from != NO_DELTAS);
+    }
```
Comment (cyx1231st) [inline src/crimson/os/seastore/cache.cc:1403]: nit:
```c++
if (unlikely(dirty_from == JOURNAL_SEQ_NULL ||
             dirty_from == JOURNAL_SEQ_MAX ||
             dirty_from == NO_DELTAS)) {
  ERRORT("extent got invalid dirty_from={} -- {}", t, dirty_from, *i);
  ceph_abort();
}
```

**Review by cyx1231st — APPROVED** [review]
LGTM

File: src/crimson/os/seastore/cache.cc (hunk 3 above)
Comment (xxhdx1985126) [reply src/crimson/os/seastore/cache.cc:1401]: done

**Review by athanatos — APPROVED** [review]

**Pull request merged**